app/TuistApp/Sources/Views/SimulatorsView/SimulatorsViewModel.swift

```diff
@@ -0,0 +1,188 @@
+
+        let urlComponents = URLComponents(url: previewURL, resolvingAgainstBaseURL: false)
+        guard let previewId = urlComponents?.queryItems?.first(where: { $0.name == "preview_id" })?.value,
+              let fullHandle = urlComponents?.queryItems?.first(where: { $0.name == "full_handle" })?.value,
+              let serverURLString = urlComponents?.queryItems?.first(where: { $0.name == "server_url" })?.value,
+              let serverURL = URL(string: serverURLString)
+        else { throw SimulatorsViewModelError.invalidDeeplink(previewURL.absoluteString) }
+
+        let downloadURL = try await downloadPreviewService.downloadPreview(
+            previewId,
+            fullHandle: fullHandle,
+            serverURL: serverURL
+        )
+
+        guard let downloadURL = URL(string: downloadURL) else { throw SimulatorsViewModelError.invalidDownloadURL(downloadURL) }
+
+        guard let archivePath = try await remoteArtifactDownloader.download(url: downloadURL)
+        else { throw SimulatorsViewModelError.appDownloadFailed(previewURL.absoluteString) }
+        let fileUnarchiver = try fileArchiverFactory.makeFileUnarchiver(for: archivePath)
+        let unarchivedDirectory = try fileUnarchiver.unzip()
+
+        let apps = try await fileHandler.glob(unarchivedDirectory, glob: "*.app").concurrentMap {
+            try await self.appBundleLoader.load($0)
+        }
+
+        guard let app = apps.first(
+            where: {
+                $0.infoPlist.supportedPlatforms.contains(
+                    where: {
+                        switch $0 {
+                        case .device:
+                            return false
+                        case let .simulator(platform):
+                            return selectedSimulator.runtime.platform == platform
+                        }
+                    }
+                )
+            }
+        )
+        else {
+            throw SimulatorsViewModelError.appNotFound(
+                selectedSimulator,
+                apps.flatMap(\.infoPlist.supportedPlatforms).compactMap {
+                    switch $0 {
+                    case .device:
+                        return nil
+                    case let .simulator(platform):
+                        return platform
+                    }
+                }
+            )
+        }
+
+        let bootedDevice = try simulatorController.booted(device: selectedSimulator.device, forced: true)
+        try simulatorController.installApp(at: app.path, device: bootedDevice)
+        try simulatorController.launchApp(bundleId: app.infoPlist.bundleId, device: bootedDevice, arguments: [])
+    }
+}
+
+extension [SimulatorDeviceAndRuntime] {
+    fileprivate func sorted() -> Self {
+        sorted(by: {
+            if $0.device.name == $1.device.name { return $0.runtime.name < $1.runtime.name }
+            else { return $0.device.name < $1.device.name }
+        })
+    }
```

Comment:
Have you considered making `SimulatorDeviceAndRuntime` [Comparable](https://developer.apple.com/documentation/swift/comparable)?

Reply:
This sorting is quite opinionated for the specific use case, so I'd rather keep it like this